sw/host/opentitanlib/src/otp/alert_handler.rs

```diff
@@ -0,0 +1,511 @@
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_BIT as usize, true);
+            }
+            AlertEnable::Locked => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_BIT as usize, true)
+            }
+        }
+
+        match config.escalate {
+            AlertEscalate::Phase0 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true)
+            }
+            AlertEscalate::Phase1 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+            }
+            AlertEscalate::Phase2 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E2_BIT as usize, true);
+            }
+            AlertEscalate::Phase3 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E2_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E3_BIT as usize, true);
+            }
+            AlertEscalate::None => {}
+        }
+
+        self.class_regs[index].ctrl = reg;
+        self.class_regs[index].accum_thresh = config.accum_thresh;
+        self.class_regs[index].timeout_cyc = config.timeout_cyc;
+        self.class_regs[index].phase_cycs = config.phase_cycs;
+
+        Ok(())
+    }
+}
+
+trait Crc32Add {
+    fn crc32_add(self, diegst: &mut Digest<u32>);
+}
+
+impl Crc32Add for u32 {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        digest.update(self.to_le_bytes().as_slice())
+    }
+}
+
+impl<T: Crc32Add, const N: usize> Crc32Add for [T; N] {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        self.map(|v| v.crc32_add(digest));
+    }
+}
+
+impl Crc32Add for AlertClassRegs {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        self.regwen.crc32_add(digest);
+        self.ctrl.crc32_add(digest);
+        self.accum_thresh.crc32_add(digest);
+        self.timeout_cyc.crc32_add(digest);
+        self.phase_cycs.crc32_add(digest);
+    }
+}
+
+impl Crc32Add for AlertRegs {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        self.regwen.crc32_add(digest);
+        self.en.crc32_add(digest);
+        self.class.crc32_add(digest);
+        self.loc_regwen.crc32_add(digest);
+        self.loc_en.crc32_add(digest);
+        self.loc_class.crc32_add(digest);
+        self.class_regs.crc32_add(digest);
+    }
+}
+
+fn new_crc() -> Crc<u32> {
+    Crc::<u32>::new(&crc::CRC_32_ISO_HDLC)
+}
+
+#[cfg(test)]
```

Comment:
@milesdai @timothytrippel can you confirm that we are running rust unit tests in the CI?

Reply:
Yes, under `Earl Grey SW Build & Test`, we are basically running `bazel test --test_tag_filters=-broken,-cw310,-verilator,-dv /sw/...`, so as long as it's registered as a bazel test, it should be run.